File: solutions/lua/sum-of-multiples/1/sum-of-multiples.lua
return function(numbers)
    return {
        to = function(level)
            local multiples, sum = {}, 0
            for _, item in ipairs(numbers) do
                for multiple = item, level - 1, item do multiples[multiple] = true end
            end
            for multiple in pairs(multiples) do sum = sum + multiple end
            return sum
        end
    }
end

File: solutions/typescript/sum-of-multiples/1/sum-of-multiples.ts
export function sum(items: number[], level: number): number {
  const multiples = new Set<number>()
  items.forEach((item) => {
    if (item <= 0)
      return
    let n = item

    while (n < level) {
      multiples.add(n)
      n += item
    }
  })
  return Array.from(multiples).reduce((next, current) => (next + current), 0)
}